docs/networking.md

```diff
@@ -0,0 +1,262 @@
+  - Ingress rules often set to 'allow all' mode (0.0.0.0/0) for ease of operations, with emphasis on consistently allow-listing required ingress traffic.
+  - Sample security groups provided as a balance between security, ease of operations, and frictionless development, with the expectation that customers will refine them based on their specific security needs.
+
+
+
+<!-- TODO: return to this
+- Difference between Security Groups and NACLs in AWS
+- Role of NACLs in providing an additional layer of security at the subnet level
+- **ASEA created and managed Security Groups and NACLs in each workload VPC**
+- **Best practices for configuring Security Groups in workload VPCs**
+  - I think this is just not to use NACLs? -->
+
+### Exposing services to the internet
+
+Generally, in the ASEA we recommend one of two methods of exposing services to the internet:
+- API Gateway
+- Application Load Balancer (ALB)
+
+Making strategic choices between AWS API Gateway and ALBs is essential for optimizing cloud architecture. API Gateway is the preferred option for internet exposure, catering to modern applications using RESTful APIs and serverless computing. ALBs are should only be used for supporting legacy applications, and require integration support from the Public Cloud team. First we'll look at an API Gateway implementation:
+
+<!-- TODO: look at the sample application for specific examples-->
+1.  **Create an API in API Gateway:**
+  - Navigate to the AWS Management Console and open the API Gateway service.
+  - Create a new API, specifying a name and description.
+  - Define the necessary resources and methods for your API, and configure the desired authentication mechanisms.
+  
+2.  **Create a VPC Link:**
+  - In the API Gateway console, navigate to the "VPC Links" section.
+  - Create a new VPC Link, specifying a name and description.
+  - Associate the VPC Link with the VPC that hosts the backend services or resources you want to access.
+
+3.  **Configure Integration with VPC Link:**
+  - In the API Gateway console, go to the API's Integration settings.
+  - Create a new HTTP integration for the resource/method you want to expose.
+  - In the integration settings, choose "VPC Link" as the integration type.
+  - Select the previously created VPC Link.
+
+4.  **Deploy the API:**
+  - Deploy the API to make it accessible via a public URL.
+  - Choose a deployment stage (e.g., production) and deploy your API.
+
+5.  **Access the Internet-Facing Service:**
+  - Once deployed, your API is accessible via the provided endpoint URL.
+  - Clients can now make requests to this endpoint, and API Gateway will route those requests to the backend services through the VPC Link.
+
+6.  **Secure the API Gateway:**
+  - Implement security measures such as API keys, IAM roles, or OAuth2 authentication depending on your requirements.
+  - Consider enabling logging and monitoring features for better visibility into API usage.
```

Comment:
Instead of putting this info directly in the doc find an official AWS doc that shows how to do this an link to it.